pkg/httphelper/httphelper.go

```diff
@@ -6,14 +6,30 @@ import (
 	"net/http"
 )
 
-func Error(w http.ResponseWriter, err error) {
+type JSONError struct {
+	Code    int         `json:"code"`
+	Message string      `json:"message"`
+	Detail  interface{} `json:"detail,omitempty"`
```

Comment:
The `Detail` field is expected to be any json serializable struct providing the API consumer with relevant information.

I don't think the `RawMessage` type is useful here as it would require consumers of `httphelper.Error` to first serialize their details object.